### ISSUE TITLE
Rails 7.1.beta.1: second test failure on system specs 

### ISSUE THREAD

**Issue by zealot128** [rspec/rspec-rails#2690]
Ruby version: 3.2.2
Rails version: 7.1.0.beta.1
RSpec version: 3.12.2

## Observed behaviour

When a failure on a system spec in a Rails app occurs (e.g. 500 error in a callback or navigation), than a second "1.2 )" failure is generated that looks like this:

```
1.2) Failure/Error: 

 `metadata` is not available from within an example (e.g. an `it` block) or from constructs that run in the scope of an example (e.g. `before` `let`, etc). It is only available on an example group (e.g. a `describe` or `context` block).
[Screenshot Image]: xxxx
```

```
gems/rspec-core-3.12.2/lib/rspec/core/example_group.rb:760:in `method_missing'
gems/activerecord-7.1.0.beta1/lib/active_record/test_fixtures.rb:278:in `method_missing'
gems/actionpack-7.1.0.beta1/lib/action_dispatch/testing/assertions/routing.rb:196:in `method_missing'
gems/actionpack-7.1.0.beta1/lib/action_dispatch/testing/integration.rb:440:in `method_missing'
gems/actionpack-7.1.0.beta1/lib/action_dispatch/system_testing/test_helpers/screenshot_helper.rb:48:in `take_failed_screenshot'
gems/actionpack-7.1.0.beta1/lib/action_dispatch/system_testing/test_helpers/setup_and_teardown.rb:8:in `before_teardown'
gems/rspec-rails-6.0.3/lib/rspec/rails/example/system_example_group.rb:110:in `call'
gems/rspec-rails-6.0.3/lib/rspec/rails/example/system_example_group.rb:110:in `block (2 levels) in <module:SystemExampleGroup>'
gems/rspec-core-3.12.2/lib/rspec/core/example.rb:457:in `instance_exec'
gems/rspec-core-3.12.2/lib/rspec/core/example.rb:457:in `instance_exec'
gems/rspec-core-3.12.2/lib/rspec/core/hooks.rb:372:in `run'
gems/rspec-core-3.12.2/lib/rspec/core/hooks.rb:529:in `block in run_owned_hooks_for'
gems/rspec-core-3.12.2/lib/rspec/core/hooks.rb:528:in `each'
gems/rspec-core-3.12.2/lib/rspec/core/hooks.rb:528:in `run_owned_hooks_for'
gems/rspec-core-3.12.2/lib/rspec/core/hooks.rb:615:in `block in run_example_hooks_for'
gems/rspec-core-3.12.2/lib/rspec/core/hooks.rb:614:in `each'
gems/rspec-core-3.12.2/lib/rspec/core/hooks.rb:614:in `run_example_hooks_for'
gems/rspec-core-3.12.2/lib/rspec/core/hooks.rb:485:in `run'
```

The culprit seems to be `take_failed_screenshot` from Rails:

```ruby
  def take_failed_screenshot
    return unless failed? && supports_screenshot? && Capybara::Session.instance_created?

    take_screenshot
    metadata[:failure_screenshot_path] = relative_image_path if Minitest::Runnable.method_defined?(:metadata)
  end
```

Which appends screenshot to their `metadata` which then gets passed upwards via `method_missing`.

## Workaround

Disabling the failure screenshots (which we never looked at anyways):

```ruby
if Rails.env.test?
  module DisableFailureScreenshots
    def take_failed_screenshot
    end
  end
  require 'action_dispatch/system_testing/test_helpers/screenshot_helper'
  ActionDispatch::SystemTesting::TestHelpers::ScreenshotHelper.prepend(DisableFailureScreenshots)
end
```

**Comment by gap777** [comment]
WAT? 

What about if we want and use that feature?

**Comment by janklimo** [comment]
This is still an issue on `rails 7.1.1` and `rspec-rails 6.0.3`.

**Comment by senid231** [comment]
my temporary fix would be:

`spec/rspec_screenshot_helper_patch.rb`
```ruby
module RspecScreenshotHelperPatch
  def metadata
    RSpec.current_example.metadata
  end
end
```

`spec/rails_helper.rb`
```ruby
require_relative 'rspec_screenshot_helper_patch'
# ...
RSpec.configure do |config|
  # ...
  config.include RspecScreenshotHelperPatch, type: :system
end
```

since in `rspec-rails` we already set current example metadata in `rspec/rails/example/system_example_group.rb`:  
```ruby
RSpec.current_example.metadata[:extra_failure_lines] = myio.readlines
```

**Comment by JonRowe** [comment]
Fix technically released in 6.0.4 but please use 6.1.0 for Rails 7.1 support.

**Comment by KirtashW17** [comment]
Is it possible that still doesn't work with Rails 7.1.3.2 and RSpec 3.13.0? I get the same error.
